app/api/routes.py

```diff
@@ -307,12 +338,18 @@ def create_resource(json, db):
     try:
         db.session.add(new_resource)
         db.session.commit()
+        index.save_object(new_resource.serialize_algolia_search)
```

Comment:
We need to add a similar call on `set_resource` (and to be honest, I don't like the name `set_resource`, we should call it `update_resource` and change the current `update_resource` to `put_resource` so that way it follows the same pattern as `post_resources()` and `create_resource`).

But of course, we should use `index.partial_update_object` and only update those fields that changed (unless its easier to just save the whole object, either way works for me)